tools/collect_debug_info.sh

```diff
@@ -0,0 +1,140 @@
+echo
+echo "Should you decide not to post the log to a gist, it will be saved"
+echo "to disk and you can scrub it before sharing it."
+echo
+echo "You are solely responsible for the data you choose to publish"
+
+source /etc/mailinabox.conf # load global vars
+TMP_FILE=/tmp/MIAB_debug_$(date -d "today" +"%Y%m%d%H%M%S").log
+
+touch $TMP_FILE;
+
+# MIAB status checks
+/root/mailinabox/management/status_checks.py >> $TMP_FILE;
+echo >> $TMP_FILE; # newline after status_checks
+
+# all of the commands we want to run.
+declare -a commands=("git -C /root/mailinabox status"
+                     "uptime"
+                     "lsb_release -d"
+                     "free -m"
+                     "df -h"
+                     "ps auxf"
+                     "pip3 list"
+                     "dpkg --list"
+                     "ufw status verbose"
+                     "ifconfig"
+                     "lsof -i"
+                     "dmesg"
+                     "cat /etc/hosts"
+                     "cat /etc/resolv.conf"
+                     "cat /var/log/syslog"
+                     "cat /var/log/mail.log"
+                     "cat /var/log/boot.log"
+                     "cat /var/log/roundcubemail/errors")
+
+function name_and_delineator () {
+  CMD_LENGTH=${#1}
+  DELINEATOR="";
+  for (( c=1; c<=$CMD_LENGTH; c++ ))
+  do
+     DELINEATOR+="=";
+  done
+  echo $1 >> $TMP_FILE
+  echo $DELINEATOR >> $TMP_FILE;
+}
+
+# iterate through each command, announce it, execute it and log it.
+for i in "${commands[@]}"
+do
+  echo "executing: $i and saving output to $TMP_FILE"
+  name_and_delineator "$i" # pretty printing
+  eval `echo $i` >> $TMP_FILE # execute string as a command, including spaces
+  echo >> $TMP_FILE; # newline
+done
+
+function post_gist () {
+  echo "Posting the debug log to https://gist.github.com at this url:"
+  echo $(gist-paste -p -s `echo $TMP_FILE`)
+  echo "Please provide this url to help diagnose your issue."
+}
+
+# double check that the user wants to post to github.
+echo "Do You want to post your debug log on https://gist.github.com?"
+echo
+echo "It will create a *secret* gist that you can delete later"
+echo
+echo "Please type 'YES' below, anything else will cancel."
+echo -n "Type YES to publish:"
+read answer
+if echo "$answer" | grep -q "^YES" ;then
+  if [ ! -f /root/.gist ]; then
+    echo "You will need to log into Github first."
+    echo "You can skip this step by pressing <ctrl-c>, but you will not"
+    echo "be able to delete the debug log if you do not log in."
+    gist-paste --login
+    # ask again, just to be sure.
+    echo "Are you sure you want to post your debug log on https://gist.github.com?"
+    echo "Please type 'YES' below, anything else will cancel."
+    echo -n "Type YES to publish:"
+    read answer
+    if echo "$answer" | grep -q "^YES" ;then
+      post_gist
+    else
+      # logged in to github, but said no on the second request
+      echo "Nothing has been posted to gist.github.com"
```

Comment:
You could wrap this bit of the else in a function and call it in both locations. 

Or check for the gist credentials separately. If you do it first then the confirmation question doesn't have to be duplicated. But I get it might be more scary for the user. 

For me this isn't a deal breaker.